sql/mdl.cc

```diff
@@ -2088,6 +2088,18 @@ MDL_context::acquire_lock(MDL_request *mdl_request, double lock_wait_timeout)
   */
   lock= ticket->m_lock;
 
+  #ifdef WITH_WSREP
+  {
+    THD *request_thd = this->get_thd();
+
+    if (request_thd->lex->sql_command == SQLCOM_CREATE_TABLE && 
+        request_thd->wsrep_exec_mode == REPL_RECV)
+    {
+      lock_wait_timeout = lock_wait_timeout ? lock_wait_timeout : LONG_TIMEOUT;
```

Comment:
`lock_wait_timeout` is 24hrs by default. How can it possibly be not enough to run `galera_concurrent_ctas` test?

Reply:
MDEV-3941 introduced problem since lock_wait_timeout will be set to 0 for this particular case.

Reply:
Can we fix caller instead?

Reply:
Fixed so that if conditions are same, lock_wait_timeout is not set to zero.

Reply:
Looks much better, thanks!